app/change_set_persisters/change_set_persister/create_proxy_file_sets.rb

```diff
@@ -2,18 +2,46 @@
 class ChangeSetPersister
   class CreateProxyFileSets
     attr_reader :change_set_persister, :change_set
+    delegate :query_service, to: :change_set_persister
+    delegate :resource, to: :change_set
 
     def initialize(change_set_persister:, change_set:)
       @change_set = change_set
       @change_set_persister = change_set_persister
     end
 
     def run
-      return unless file_set_ids.present?
-      change_set.member_ids += proxy_file_set_ids
+      if detached_proxy_file_set_ids.present?
+        delete_detached_proxy_file_sets
+        change_set.member_ids = attached_proxy_file_set_ids
+      else
+        return if file_set_ids.nil?
+        change_set.member_ids += proxy_file_set_ids
+      end
+
       change_set.sync
     end
 
+    def detached_proxy_file_set_ids
+      change_set.try(:detached_member_ids)
```

Comment:
Why is this all wrapped up in the same hook? Also, why does it need to be built into this at all - can't we just use the same delete logic we've always used for members?